flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateCheckpointWriter.java

```diff
@@ -296,13 +296,17 @@ private void runWithChecks(RunnableWithException r) throws Exception {
             r.run();
         } catch (Exception e) {
             fail(e);
-            throw e;
         }
     }
 
     public void fail(Throwable e) throws Exception {
```

Comment:
Good point, thanks.

Though I'm going to partially revert this change: otherwise, programmatic and other errors aren't propagated (that's why `ChannelStateWriteRequestDispatcherTest` fails). But they don't have to be declared (non-recoverable by definition).

Instead, I'm going to add something like
```
            fail(e);
            if (!ExceptionUtils.findThrowable(e, IOException.class).isPresent()) {
                ExceptionUtils.rethrow(e);
            }
```
(but first I'd like to confirm that the original exception was indeed `IOException`)